### PR TITLE
export query results from collections

### PR DESCRIPTION
**Pull request by schuttsm**
Addresses: https://github.com/mongo-express/mongo-express/issues/385

* add export collection query results to json button
* change export collection api call to filter based on query

### CODE REVIEW

File: lib/routes/collection.js [hunk 1]
```diff
@@ -119,7 +115,10 @@ var routes = function (config) {
       limit:  limit,
       skip:   skip,
     };
+    return query_options;
+  };
 
+  exp._getQuery = function (req, res) {
```
Comment (MikeyBurkman) [inline lib/routes/collection.js]: I like having these new functions, but the error case when parsing the query in this function might cause issues. Before, the `return` in the case of a bad query would stop execution of the viewCollection() function, but now it'll just stop execution of this new function, and `query` will be some nonsense value on line 147. The session will redirect, but the query will still run (maybe) and who know what happens after that.

Might have to just let the error propagate up, surround line 147 with a try/catch, and then handle the error and redirect there.

(That being said, though it's almost certainly out of scope here, I'd love for the app to restore the query/projection values when there's an error parsing the query.)

Comment (schuttsm) [reply lib/routes/collection.js]: @MikeyBurkman - good catch!

I moved the try catch blocks up a level so the parent functions are returned on error.

**Comment by MikeyBurkman** [comment]
Alright, so I ran this locally and it worked fine. The only maybe-issue I see is that it exports only as the standard format, while I personally was expecting JSON array. It's of course not wrong (my intuition was wrong), but it makes me wonder if we can make this even better.

I feel like the next logical step after this would be to allow exporting the current query in the other formats. @dozoisch What do you think? Instead of adding a new export button, maybe just make the current export buttons export what you've queried? It's a change in current behavior, but I feel it's more intuitive that way.

I think the hard work is already done--we would just need to use the same query/project/query_options in the `collection.find()` calls in `exportColArray` and `exportCsv`, and of course pass in those options to each button via the html, like you're already doing.

**Comment by schuttsm** [comment]
@MikeyBurkman - thank you for the feedback.

I removed the new export button & hooked the query filter & projection inputs into the existing export buttons and you were right - so much more intuitive.  How does it look now?

**Comment by MikeyBurkman** [comment]
@schuttsm Thanks!

**Pull request merged**